trivup/apps/OauthbearerOIDCApp.py

```diff
@@ -0,0 +1,149 @@
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from trivup import trivup
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import jwt
+import datetime
+import json
+import argparse
+from threading import Thread
+import requests
+
+class WebServerHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text')
+        self.end_headers()
+        message = "HTTP server for OAuth\n"
+        message += "Example for token retrieval:\n"
+        message += 'curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "accept: application/json" \
+        -H "authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "content-type: application/x-www-form-urlencoded" \
+        -d "method=oidc,scope=test-scope"'
+        self.wfile.write(message.encode())
+        print (message)
+
+    def do_POST(self):
+        """
+        Example usage:
+        curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "accept: application/json" \
+        -H "authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "content-type: application/x-www-form-urlencoded" \
+        -d "method=oidc,scope=test-scope"
+        """
+        if not self.path.endswith("/retrieve"):
+            self.send_error(404, 'URL is not valid: %s' % self.path)
+            return
+
+        if self.headers.get('Content-Length', None) is None: {
+            self.send_error(404, 'Content-Length field is required')
+        }
+
+        content_length = int(self.headers['Content-Length'])
+        post_data = self.rfile.read(content_length)
+
+        if self.headers.get('Content-Type', None) is None: {
+            self.send_error(404, 'Content-Type field is required')
+        }
+
+        if self.headers.get('Authorization', None) is None: {
+            self.send_error(404, 'Authorization field is required')
+        }
+
+        if post_data is None:
+            self.send_error(404,
+                            'method and scope fields are required in data')
+
+        payloads = {"exp": datetime.datetime.utcnow() + \
+                           datetime.timedelta(seconds=30)}
+
+        encoded_jwt = jwt.encode(payloads, "secret", algorithm="HS256",
+                                 headers={"kid":
+                                          self.headers['authorization']})
+        self.send_response(200)
+        self.send_header('Content-type', 'text')
+        self.end_headers()
+        messages = {"access_token": encoded_jwt}
+        self.wfile.write(json.dumps(messages, indent = 4).encode())
+
+class OauthbearerOIDCHttpServer():
+    def enable_http_server(self, port):
+        server = HTTPServer(('', port), WebServerHandler)
+        thread = Thread(target=server.serve_forever, args=())
+        thread.start()
```

Comment:
With this thread in the background, what is keeping this process alive, isn't it reaching the end of __main__ after this and exiting?
I'm thinking you should perhaps run serve_forever() here instead of using a thread, but I'm not sure.